### DISCUSSION TITLE
pyintaller generated exe file gives error

### DISCUSSION THREAD

**Discussion by smehra69** [pyinstaller/pyinstaller#5781] [Help] [answered]
My program works fine if I run the program.py file in a python environment. 
The command used was:-
pyinstaller --noconfirm --onedir -D program.py

Any code, even a single, one line print command throws this error
Then i use pyinstaller to create an executable using onedir option, 
When this exe file is run in a different PC(Not having python installed) I get an error. What can I do to set it right? The error is:- 

Traceback (most recent call last): 
  File "Lib\site-packages\pyinstaller-5.0.dev0-py3.9.egg\PyInstaller\hooks\rthooks\pyi_rth_multiprocessing.py", line 17, in <module>
  File "PyInstaller\loader\pyimod03_importers.py", line 531, in exec_module 
  File "multiprocessing\__init__.py", line 16, in <module>
  File "PyInstaller\loader\pyimod03_importers.py", line 531, in exec_module 
  File "multiprocessing\context.py", line 6, in <module>
  File "PyInstaller\loader\pyimod03_importers.py", line 531, in exec_module 
  File "multiprocessing\reduction.py", line 16, in <module>
  File "PyInstaller\loader\pyimod03_importers.py", line 531, in exec_module 
  File "socket.py", line 51, in <module>
ModuleNotFoundError: No module named '_socket' 
[11744] Failed to execute script pyi_rth_multiprocessing

**Comment by rokm** [comment]
Are you by any chance using the executable from `build` directory instead of the one from `dist` (like #5529 did)?

**Comment by smehra69** [comment]
No. using from dist directory only.

**Comment by rokm** [comment]
Does the application's dist directory contain any `.pyd` files?

**Comment by smehra69** [comment]
This is the \dist\test1 directory listing. Yes there are .pyd files

27-04-2021  16:50           776,948 base_library.zip
27-04-2021  16:41    <DIR>          Include
03-03-2021  11:39         3,399,200 libcrypto-1_1.dll
03-03-2021  11:39            32,792 libffi-7.dll
03-03-2021  11:39           689,184 libssl-1_1.dll
03-03-2021  11:39           188,944 pyexpat.pyd
22-02-2021  17:35         4,457,488 python39.dll
22-02-2021  17:35            27,152 select.pyd
27-04-2021  16:50         2,028,817 test1.exe
27-04-2021  16:50             1,493 test1.exe.manifest
22-02-2021  17:35         1,119,760 unicodedata.pyd
22-02-2021  17:35           101,672 VCRUNTIME140.dll
03-03-2021  11:39            63,504 _asyncio.pyd
03-03-2021  11:39            85,008 _bz2.pyd
03-03-2021  11:39           270,352 _decimal.pyd
03-03-2021  11:39           161,296 _lzma.pyd
22-02-2021  17:35            28,688 _multiprocessing.pyd
03-03-2021  11:39            45,072 _overlapped.pyd
03-03-2021  11:39            28,176 _queue.pyd
              18 File(s)     13,505,546 bytes

**Comment by rokm** [comment]
Hm, can you try adding `--hiddenimport _socket` to your pyinstaller command and see if `_socket.pyd` gets collected?

**Comment by smehra69** [comment]
Did hiddenimport for _socket. Error still there.

**Comment by bwoodsend** [comment, marked as answer]
Can you run:
```python
python -c "from _socket import __file__; print(__file__)"
```
And tell us what the output is?

**Comment by smehra69** [comment]
Output for python -c "from _socket import __file__; print(__file__)" is 
D:\PYTHON\DLLs\_socket.pyd
